### PR TITLE
Create first (simple) end-to-end and integration tests

### PR DESCRIPTION
**Pull request by JohanLorenzo**
WIP

### CODE REVIEW

File: tests/e2e/lib/views/login/accessors.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  get usernameField() {
+    return this.waitForElement(By.css('input[type="text"]'));
+  },
+
+  get passwordField() {
+    return this.waitForElement(By.css('input[type="password"]'));
+  },
+
+  get submitButton() {
+    return this.waitForElement(By.css('button'));
```
Comment (JohanLorenzo) [inline tests/e2e/lib/views/login/accessors.js]: We could probably have a better selector for it. What do you think @gmarty ?

File: tests/e2e/lib/views/login/accessors.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+var _ = require('lodash');
+
+
+function LoginAccessors() {
+  Accessors.apply(this, arguments);
+}
+
+LoginAccessors.prototype = _.assign({
+  get usernameField() {
+    return this.waitForElement(By.css('input[type="text"]'));
```
Comment (JohanLorenzo) [inline tests/e2e/lib/views/login/accessors.js]: We could probably have a better selector for it. What do you think @gmarty ?

**Comment by JohanLorenzo** [comment]
Here's the first revision of an E2E test. There are a couple of things that have to be addressed: 
* we require some manual set up to make them run
* No CI runs the tests, yet (because of the previous point). As a consequence, the test might bitrot.

I'd propose to handle these issues in a follow up, for 2 main reasons:
* Executing E2E tests on travis will require a fair amount of effort in set up (clone repos, make builds, run the server and go through the set up app). This'll make the PR heavier to review
* The login test can be executed against a fake foxbox In this case we require to create that fake server. Also some heavy work to review.

I'd suggest to first deal with the fake server, which looks simpler to deal with. Especially, since we have something equivalent in fxbox/foxbox#176.

r? @gmarty 

cc @isabelrios @npark-mozilla

File: deploy.sh [hunk 3]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash
 # Deploys to GitHub Pages.
-# Make sure to stash your local changes before running this script.
```
Comment (gmarty) [inline deploy.sh]: There's a npm postinstall script that revert some files. I need to fix that but in the meantime, please revert deploy.sh and gulpfile.js or don't commit the changes.

**Comment by gmarty** [comment]
The login mechanism is currently being changed. It's now redirect to the box to get a token back.
Can you please wait until the new auth lands before merging this branch?

**Comment by JohanLorenzo** [comment]
Updated to use the new login flow. Test is now passing on Travis. r? @gmarty

File: README.md [hunk 4]
```diff
@@ -58,3 +58,13 @@ cordova plugin remove com.synconset.cordovaHTTP
 cordova plugin add /path/to/local/SecureHTTP
 cordova run android
 ```
+## Tests
+### All regular tests
+Run `npm test`.
+
+### Run end-to-end tests (e2e)
```
Comment (gmarty) [inline README.md]: Please breaklines before and after headers.

Comment (JohanLorenzo) [inline README.md]: Done

**Comment by gmarty** [comment]
Can you squash all your commits into a single one?
The rest looks good to me!

**Comment by JohanLorenzo** [comment]
Commits squashed and build still green. Merging.

**Pull request merged**